test/test_video.py

```diff
@@ -0,0 +1,224 @@
+    #     # PyAv miss one audio frame at the beginning (pts=0)
+    #     check_aframes=False,
+    #     check_aframe_pts=False,
+    # ),
+    # "SOX5yA1l24A.mp4": GroundTruth(
+    #     duration=11.0,
+    #     video_fps=29.97,
+    #     audio_sample_rate=48000,
+    #     # PyAv miss one audio frame at the beginning (pts=0)
+    #     check_aframes=False,
+    #     check_aframe_pts=False,
+    # ),
+    # "WUzgd7C1pWA.mp4": GroundTruth(
+    #     duration=11.0,
+    #     video_fps=29.97,
+    #     audio_sample_rate=48000,
+    #     # PyAv miss one audio frame at the beginning (pts=0)
+    #     check_aframes=False,
+    #     check_aframe_pts=False,
+    # ),
+}
+
+def _template_read_video(video_object, s=0, e=None):
+    
+    if e is None:
+        e = float("inf")
+    if e < s:
+        raise ValueError(
+            "end time should be larger than start time, got "
+            "start time={} and end time={}".format(s, e)
+        )
+
+    video_object.set_current_stream("video")
+    video_object.seek(s)
+    video_frames = torch.empty(0)
+    frames = []
+    t, pts = video_object.next()
+    while t.numel() > 0 and (pts >= s and pts <= e):
+        frames.append(t)
+        t, pts = video_object.next()
+    if len(frames) > 0:
+        video_frames = torch.stack(frames, 0)
+
+    video_object.set_current_stream("audio")
+    video_object.seek(s)
+    audio_frames = torch.empty(0)
+    frames = []
+    t, pts = video_object.next()
+    while t.numel() > 0 and (pts > s and pts <= e):
+        frames.append(t)
+        t, pts = video_object.next()
+    if len(frames) > 0:
+        audio_frames = torch.stack(frames, 0)
+
+    return video_frames, audio_frames, video_object.get_metadata()
+
+@unittest.skipIf(_HAS_VIDEO_OPT is False, "Didn't compile with ffmpeg")
+class TestVideo(unittest.TestCase):
+    def test_read_video_tensor(self):
+        """
+        Check if reading the video using the `next` based API yields the
+        same sized and equal tensors as video_reader.
+        """
+        torchvision.set_video_backend("video_reader")
+        for test_video, config in test_videos.items():
+            full_path = os.path.join(VIDEO_DIR, test_video)
+            # pass 1: decode all frames using existing TV decoder
+            tv_result, _, _ = torchvision.io.read_video(full_path, pts_unit="sec")
+            tv_result = tv_result.permute(0, 3, 1, 2)
+            # pass 2: decode all frames using new api
+            reader = torch.classes.torchvision.Video(full_path, "video", True)
+            frames = []
+            t, _ = reader.next()
+            while t.numel() > 0:
+                frames.append(t)
+                t, _ = reader.next()
+            new_api = torch.stack(frames, 0)
+            self.assertEqual(tv_result.size(), new_api.size())
+            self.assertEqual(torch.equal(tv_result, new_api), True)
+    
+    @unittest.skipIf(not _HAS_VIDEO_OPT, "video_reader backend is not chosen")
```

Comment:
we don't need `skipIf` for every test if we already added it in the class constructor. The same comment applies to all other methods in this file